File: seed/app/src/main/java/com/WATER_ME_WITH_COMPANY_GOODNESS/android/WATER_ME_WITH_APP_NAME/SimpleAppLifecycleCallbacks.java
package com.{company_name}.android.{app_package_name_prefix};

import android.app.Activity;
import android.app.Application;
import android.os.Bundle;

/**
 * Empty implementation of {@link android.app.Application.ActivityLifecycleCallbacks}
 * to allow base classes to only implement the methods they're interested in.
 */
abstract class SimpleAppLifecycleCallbacks implements Application.ActivityLifecycleCallbacks {

    @Override
    public void onActivityCreated(Activity activity, Bundle savedInstanceState) {

    }

    @Override
    public void onActivityStarted(Activity activity) {

    }

    @Override
    public void onActivityResumed(Activity activity) {

    }

    @Override
    public void onActivityPaused(Activity activity) {

    }

    @Override
    public void onActivityStopped(Activity activity) {

    }

    @Override
    public void onActivitySaveInstanceState(Activity activity, Bundle outState) {

    }

    @Override
    public void onActivityDestroyed(Activity activity) {

    }
}

File: garden/seeds/kotlin/app/src/main/kotlin/com/WATER_ME_WITH_COMPANY/android/WATER_ME_WITH_APP_NAME/util/SimpleAppLifecycleCallbacks.kt
package com.{{company_name}}.android.{{app_package_name_prefix}}.util

import android.app.Activity
import android.app.Application
import android.os.Bundle

/**
 * Empty implementation of [android.app.Application.ActivityLifecycleCallbacks]
 * to allow base classes to only implement the methods they're interested in.
 */
abstract class SimpleAppLifecycleCallbacks : Application.ActivityLifecycleCallbacks {

    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
    }

    override fun onActivityStarted(activity: Activity) {
    }

    override fun onActivityResumed(activity: Activity) {
    }

    override fun onActivityPaused(activity: Activity) {
    }

    override fun onActivityStopped(activity: Activity) {
    }

    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle?) {
    }

    override fun onActivityDestroyed(activity: Activity) {
    }
}
